daft/execution/execution_step.py

```diff
@@ -629,12 +629,21 @@ class Join(SingleOutputInstruction):
     left_on: ExpressionsProjection
     right_on: ExpressionsProjection
     how: JoinType
+    is_swapped: bool
 
     def run(self, inputs: list[MicroPartition]) -> list[MicroPartition]:
         return self._join(inputs)
 
     def _join(self, inputs: list[MicroPartition]) -> list[MicroPartition]:
-        [left, right] = inputs
+        *lefts, right = inputs
```

Comment:
Is there not an invariant here that `len(inputs) == 2`?

Reply:
Ah I see, the `left` is potentially broadcasted partition(s).

Maybe add a small comment in here somewhere?